scripts/auto-update-release-index.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+import re
+import subprocess
+
+import ruamel.yaml
+from ruamel.yaml.main import YAML
+
+
+def get_toplevel() -> str:
+    return subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"], capture_output=True, text=True, check=True
+    ).stdout.strip()
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser("Automatic update of release index")
+    parser.add_argument("commit", help="Commit of the release, it will be used to determine the rc")
+    parser.add_argument("link", help="Link to the pipeline of qualification job")
+    return parser.parse_args()
+
+
+def get_branch_with_commit(commit: str) -> str:
+    output = subprocess.run(
+        ["git", "ls-remote", "--branches", "https://github.com/dfinity/ic.git"],
+        capture_output=True,
+        text=True,
+        check=True,
+    ).stdout.strip()
+
+    branch = list(filter(lambda line: line.startswith(commit), output.splitlines()))[0].split("/")[-1]
+    return branch
+
+
+def parse_branch(branch: str) -> tuple[str, str]:
+    pattern = re.compile(r"^(rc--\d{4}-\d{2}-\d{2}_\d{2}-\d{2})(?:-(.*))?$")
+    matches = pattern.match(branch)
+    if not matches:
+        raise ValueError("Input branch is not in the correct format `rc--%Y-%m-%d_%H-%M(-feature)")
+
+    name = matches.group(1)
+    feature = matches.group(2)
+    if not feature:
+        feature = "base"
+
+    return (name, feature)
+
+
+def check_if_should_pop_latest_rc(rc_name: str, index):
+    output = subprocess.run(["git", "diff", "main", "--", "release-index.yaml"], capture_output=True, text=True)
```

Comment:
not sure if this will always work like you expect. i think git diff still might include this release even if it's present on both branches. it's for sure correct to just checkout file on main and determine if it's already present there or not.

Reply:
Will test it out and adapt if needed. Thanks for suggestion